File: VeraPelle/src/main/java/psw/verapelle/DTO/OrderDTO.java
package psw.verapelle.DTO;

import psw.verapelle.entity.OrderStatus;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderDTO {

    private Long id;
    private LocalDateTime date;
    private OrderStatus status;
    private BigDecimal totalAmount;
    private String shippingAddress;

    /**
     * Lista dettagli item
     */
    private List<OrderItemDetailDTO> items;

    /**
     * Stato simulato del pagamento ("APPROVED" o "DECLINED")
     */
    private String paymentStatus;
}

File: VeraPelle_front_standalone/src/app/core/models/order-dto.ts
// src/app/models/order-dto.ts

export enum OrderStatus {
  CREATED = 'CREATED',
  PAID = 'PAID',
  DECLINED = 'DECLINED',
  CANCELLED = 'CANCELLED'
}

export interface OrderItemDetailDTO {
  productId: number;
  productName: string;
  quantity: number;
  unitPrice: number;
}

export interface OrderDTO {
  id: number;
  date: string;              // ISO string dal backend
  status: OrderStatus;
  totalAmount: number;
  shippingAddress: string;
  items: OrderItemDetailDTO[];
  paymentStatus: 'APPROVED' | 'DECLINED';
}
